src/Selenium2Library/keywords/_element.py

```diff
@@ -649,28 +649,63 @@ def xpath_should_match_x_times(self, xpath, expected_xpath_count, message='', lo
                    % (actual_xpath_count, xpath))
 
     # Public, custom
-    def add_location_strategy(self, strategy_name, strategy_keyword, persist=False):
-        """Adds a custom location strategy based on a user keyword. Location strategies are
-        automatically removed after leaving the current scope by default. Setting `persist`
-        to any non-empty string will cause the location strategy to stay registered throughout
-        the life of the test.
-
-        Trying to add a custom location strategy with the same name as one that already exists will
-        cause the keyword to fail.
-
-        Custom locator keyword example:
-        | Custom Locator Strategy | [Arguments] | ${browser} | ${criteria} | ${tag} | ${constraints} |
-        |   | ${retVal}= | Execute Javascript | return window.document.getElementById('${criteria}'); |
-        |   | [Return] | ${retVal} |
-
-        Usage example:
-        | Add Location Strategy | custom | Custom Locator Strategy |
-        | Page Should Contain Element | custom=my_id |
+    def add_location_strategy(self, strategy_name, strategy_keyword=None, location=None, persist=False):
```

Comment:
This is backwards incompatible. `location` should be added to the end of the argument list.